ice/recipes/run_over_gs.py

```diff
@@ -0,0 +1,34 @@
+import pandas as pd
+from ice.recipe import Recipe, recipe
+from ice.evaluation.evaluate_recipe_result import RecipeResult
+from ice.utils import map_async
+from ice.evaluation.evaluation_report import EvaluationReport
+from ice.evaluation.evaluate_recipe_result import EvaluatedRecipeResult
+
+def make_recipe_result(row: pd.Series) -> RecipeResult:
+    return RecipeResult(
+        question_short_name=row.question_short_name,
+        document_id=row.get("document_id", ""),
+        answer="" if pd.isna(row.answer) else row.answer,
+        experiment=row.get("experiment", ""),
+        excerpts=row.get("excerpts", []),
+    )
+
+async def run_recipe_on_row(row: pd.Series, recipe_to_run: Recipe):
+    return await recipe_to_run(**row)
+
+async def run_over_gs(recipe_to_run: Recipe, gs_df: pd.DataFrame) -> EvaluationReport:
+    gs_df["answer"] = await map_async(
+        [row for _, row in gs_df.iterrows()],
+        lambda row: run_recipe_on_row(row, recipe_to_run),
```

Comment:
not sure if this is the most elegant way to handle an "async lambda", didn't think about it very hard